modules/w25qxx/README.md

```diff
@@ -0,0 +1,114 @@
+4. Enter <pre>w <i>string</i></pre> into the terminal, to save the string
+into flash memory
+5. Once confirmation appears on the Serial console, disconnect the board and power it off
+6. Since flash memory is secondary memory, the data should remain on the board
+7. Connect the board once again, and open Serial Monitor
+8. Enter `r` in the monitor, to request the development board to read the data stored
+9. Verify that the data retrieved matches the data that was entered
+
+## Code Explanation
+
+A few things to note about the SPI Flash memory module:
+- Since the module is a secondary storage module, it should retain information
+written to it even when power is disconnected
+- The flash module requires that to write to a certain address in the module, 
+the existing data in that address must be erased initially
+
+
+```arduino
+SPIClass hspi(HSPI);
+SPIFlash flash(15, &hspi);;
+
+uint32_t addr = 0x123;
+```
+The ESP32 has 4 SPI peripherals, SPI0, SPI1, HSPI and VSPI. <br>
+SPI0 and SPI1 are used internally, to communicate to the built-in flash memory.
+Our development board connects peripherals to HSPI.
+
+`SPIClass` is a class which represents an SPI interface. It's constructore takes
+the descriptor for that interface, which in this case has been provided by the
+ESP32 library.
+
+`SPIFlash` is the class which actually handles the flash operations. The arguments for its constructor
+are the slave select pin number and the address of the `SPIClass` it's required to used.
+
+`addr` is the address within the flash memory, to be used for read/write operations.
+It has been set to an arbitrary value of `0x123`, feel free to modify this value,
+to read/write data to a different address.
+
+```arduino
+void readStr() {
+        String s;
+        if (flash.readStr(addr, s)) {
+                Serial.println("Read '" + s + "'");
+  } else {
+                Serial.println("ERROR: Couldnt read array");
+        }
+}
+```
+
+`readStr` is a utility function to read from the flash memory. It makes use of the
+`readStr` function provided by the `flash` object. Arguments to this function
+are the address to read from, and the string to store the information in.
+This function returns `TRUE` if the read operation succeeded and `FALSE` otherwise.
+Both cases have been handled in this scenario, and the user will be intimated
+if there is an error
+
+```arduino
+void writeStr(String s) {
+        s.trim();
+  flash.eraseSection(addr, s.length() + 1);
+  Serial.println("Writing '" + s + "' ...");
+        if (flash.writeStr(addr, s)) {
+                Serial.println("LOG: Done");
+        } else {
+                Serial.println("ERROR: Couldnt write");
+        }
+}
+```
+`writeStr` is a utility function similar to `readStr`, responsible for writing a string.
+1. `s.trim()` removed spaces from the left and right ends of the string
+2. `flash.eraseSelection(addr, s.length() + 1)` erases enough bytes in the flash 
```

Comment:
Alter the sentence structure. 
Probably something like : Starts from the address to which we plan to write and erases sufficient number of bytes in the flash memory to store the string. 

Feel free to come up with something better :)